File: 1927-sum-game/1927-sum-game.java
class Solution {
    public boolean sumGame(String num) {
        int n = num.length();

        int sumLeft = 0;
        int sumRight = 0;

        int qLeft = 0;
        int qRight = 0;

        for (int i = 0; i < n / 2; i++) {
            if (num.charAt(i) == '?') {
                qLeft++;
            } else {
                sumLeft += num.charAt(i) - '0';
            }
        }

        for (int i = n / 2; i < n; i++) {
            if (num.charAt(i) == '?') {
                qRight++;
            } else {
                sumRight += num.charAt(i) - '0';
            }
        }

        if (qLeft == qRight) {
            return sumLeft != sumRight;
        }

        int qDiff = qLeft - qRight;
        int sumDiff = sumLeft - sumRight;

        return sumDiff * 2 != -9 * qDiff;
    }
}

File: 1927-sum-game/1927-sum-game.py
class Solution:
    def sumGame(self, num: str) -> bool:
        n = len(num)

        sum_left = 0
        sum_right = 0

        q_left = 0
        q_right = 0

        for i in range(n // 2):
            if num[i] == '?':
                q_left += 1
            else:
                sum_left += int(num[i])

        for i in range(n // 2, n):
            if num[i] == '?':
                q_right += 1
            else:
                sum_right += int(num[i])

        if q_left == q_right:
            return sum_left != sum_right

        q_diff = q_left - q_right
        sum_diff = sum_left - sum_right

        return sum_diff * 2 != -9 * q_diff